lib/constants.py

```diff
@@ -18,7 +18,7 @@
 import os
 
 # Images which are OSTree based
-OSTREE_IMAGES = ["centos-9-bootc", "fedora-coreos", "rhel4edge"]
+OSTREE_IMAGES = ("centos-9-bootc", "fedora-coreos", "rhel4edge")
```

Comment:
OOI, is that really a typing issue, or  a random unrelated change? I know you have a preference for tuples, but I think conceptually they are the wrong data type for this kind of use case: It is really intended that all entries have the same type, which isn't the case for tuples.

Reply:
You can view this from two sides, I think.

The first is from the dynamic/pragmatic/implementation side.  In that case, a tuple is essentially exactly the same as an array except that:
 - all of the methods for modifying it are removed
 - `[] + ()` doesn't produce the same result as `[] + []` (since Python doesn't know what type to Pick, but you should probably write that as `[*[], *()] anyway.
Also, if you look around online for "immutable list Python" everyone says "use a tuple".  This is pervasive practice.

The second side is from the static/type-theoretical standpoint, which I think is the one you're arguing.  I agree that tuples and list are not semantically the same thing, and this part of it also looks odd to my eye, but Python has a very strong tradition of using tuples also as immutable lists.  Since the beginning it was always valid to write `Tuple[str, ...]` to mean "immutable list of strings" (and now, of course, it's also valid to write `tuple[str, ...]`).

Would you feel better if we annotated that as `tuple[str, ...]`?  We could also type it as a `Sequence` or `Collection`...  We could also change it to a `frozenset`.  For me, the main concern is immutability.

We've had actual bugs in our code before where someone does something like:

```python
  myvalue = SOME_CONSTANT
  if condition:
    myvalue += ['foo']
```

without realizing that they're modifying `SOME_CONSTANT`...